packages/@aws-cdk/aws-ec2/lib/bastion-host.ts

```diff
@@ -0,0 +1,76 @@
+import { PolicyStatement } from "@aws-cdk/aws-iam";
+import { CfnOutput, Construct } from "@aws-cdk/core";
+import { AmazonLinuxGeneration, AmazonLinuxImage, InstanceClass, InstanceSize, InstanceType } from ".";
+import { Instance } from "./instance";
+import { Port } from "./port";
+import { IVpc, SubnetType } from "./vpc";
+
+/**
+ * Properties of the bastion host
+ */
+export interface BastionHostProps {
+
+    /**
+     * In which AZ to place the instance within the VPC
+     *
+     * @default - Random zone.
+     */
+    readonly availabilityZone?: string;
+
+    /**
+     * VPC to launch the instance in.
+     */
+    readonly vpc: IVpc;
+
+    /**
+     * The name of the instance
+     *
+     * @default - CDK generated name
+     */
+    readonly instanceName?: string;
+
+    /**
+     * Use public subnet instead of private one
+     *
+     * @default - false
+     */
+    readonly publicSubnets?: boolean;
+
+    /**
+     * Allow classic SSH access via port 22 instead of SSM
+     *
+     * @default - false
+     */
+    readonly allowClassicSSH?: boolean;
+}
+
+export class BastionHost extends Instance {
+    constructor(scope: Construct, id: string, props: BastionHostProps) {
+        super(scope, id, {
+            vpc: props.vpc,
+            availabilityZone: props.availabilityZone,
+            instanceName: props.instanceName || 'BastionHost',
+            instanceType: InstanceType.of(InstanceClass.T3, InstanceSize.NANO),
+            machineImage: new AmazonLinuxImage({ generation: AmazonLinuxGeneration.AMAZON_LINUX_2 }),
+            vpcSubnets: props.publicSubnets ? { subnetType: SubnetType.PUBLIC } : { subnetType: SubnetType.PRIVATE },
+        });
+        this.addToRolePolicy(new PolicyStatement({
+            actions: [
+                'ssmmessages:*',
+                'ssm:UpdateInstanceInformation',
+                'ec2messages:*'
+            ],
+            resources: ['*'],
+        }));
+        this.addUserData('yum install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm');
+
+        if (props.allowClassicSSH) {
+            this.connections.allowFromAnyIpv4(Port.tcp(22), 'Allow SSH access from everywhere');
```

Comment:
I think this may be a bad idea; the end-user would be opening this instance up to the whole world by default.
The AWS Well-Architected Framework Security Pillar advises against,
There is also a rule in the CIS AWS Foundation to detect such security groups. ( Can be found in SecurityHub)
4.1 Ensure no security groups allow ingress from 0.0.0.0/0 to port 22

Reply:
I fully agree with this objection. On the other hand, this is exactly what I see with many AWS users. If the server is up to date and key management is working correctly, it should not be a problem to have 22 open and it is the only way if your clients have dynamic IPs from their ISPs.

Reply:
How about requiring the user to explicitly enable this via a setting which will be `false` by default?